cmdstanpy/model.py

```diff
@@ -655,9 +655,48 @@ def run_generated_quantities(
 
         :return: CmdStanGQ object
         """
-        generate_quantities_args = GenerateQuantitiesArgs(csv_files=csv_files)
-        generate_quantities_args.validate(len(csv_files))
-        chains = len(csv_files)
+        sample_csv_files = []
+        sample_drawset = None
+        chains = 0
+
+        if isinstance(mcmc_sample, CmdStanMCMC):
+            sample_csv_files = mcmc_sample.runset.csv_files
+            sample_drawset = mcmc_sample.get_drawset()
+            chains = mcmc_sample.chains
+        elif isinstance(mcmc_sample, list):
+            sample_csv_files = mcmc_sample
+        else:
+            raise ValueError(
+                'mcmc_sample must be either CmdStanMCMC object'
+                ' or list of paths to sample csv_files'
+            )
+
+        try:
+            chains = len(sample_csv_files)
+            if sample_drawset is None:  # assemble sample from csv files
+                sampler_args = SamplerArgs()
+                args = CmdStanArgs(
+                    self._name,
+                    self._exe_file,
+                    chain_ids=[x + 1 for x in range(chains)],
+                    method_args=sampler_args,
+                )
+                runset = RunSet(args=args, chains=chains)
+                runset._csv_files = sample_csv_files
+                sample_fit = CmdStanMCMC(runset)
+                sample_fit._validate_csv_files()
+                sample_drawset = sample_fit.get_drawset()
+        except Exception:
```

Comment:
Only problem with these are, that it is really hard to debug what really failed.

Reply:
made error message more verbose and includes messages from the utils functions which scan/parse the csv files.